src/libs/SearchUtils.ts

```diff
@@ -335,10 +334,33 @@ function buildJSONQuery(query: string) {
     }
 }
 
-function getFilters(query: string, fields: Array<Partial<AllFieldKeys>>) {
-    let jsonQuery;
+function buildSearchQueryString(partialQueryJSON?: Partial<SearchQueryJSON>) {
+    const queryParts: string[] = [];
+    const defualtQueryJSON = buildSearchQueryJSON('');
+
+    // For this const values are lowercase version of the keys. We are using lowercase for ast keys.
+    for (const [, value] of Object.entries(CONST.SEARCH.SYNTAX_ROOT_KEYS)) {
+        if (partialQueryJSON?.[value]) {
+            queryParts.push(`${value}:${partialQueryJSON[value]}`);
+        } else if (defualtQueryJSON) {
+            queryParts.push(`${value}:${defualtQueryJSON[value]}`);
+        }
+    }
+
+    return queryParts.join(' ');
+}
+
+// Fill query string with all default params.
+function normalizeQuery(query: string) {
+    const normalizedQueryJSON = buildSearchQueryJSON(query);
+    return buildSearchQueryString(normalizedQueryJSON);
+}
```

Comment:
Why do we need this function? I think that it could inject params into the query string that were not defined by the user and that might mess up with how we display the filter UI, no?

Reply:
This is something we could discuss more about. We have a couple of questions here:
1. Do we want to put defaults for sort etc in the URL? (I think yes, they were here in the previous version). 
2. Do we want to `organize` the query for the user? So is the:
`type:expenses sortBy:date` same query as `sortBy:date type:expenses` I would say yes. In that case we should sort the keywords in the query to make it look the same.

And to address your concerns. It will inject defaults but nothing more. It shouldn't break UI filters.

Another argument for normalizing but it will be relevant in the future. If we ever want to change the defaults it could break the saved queries if we don't normalize them.

Reply:
>Do we want to put defaults for sort etc in the URL? (I think yes, they were here in the previous version).

Thinking a bit ahead, we'd only want to do that if the user explicitly types the defaults in their queries, otherwise it'd be weird for the user to type `amount>200` and then see the search bar update to `type:expense status:all amount>200 sortBy:date sortOrder:desc offset:0`. I don't think we should modify the user input in that way and the URL is kinda the only thing tying everything up, e.g. the user input, the UI filters and the syntax itself. I think we'd only want to normalize it/build the AST when hashing and sending the API request

Reply:
This is NAB for now, but something that we should align on before we get too deep in it

Reply:
I agree with Adam and I feel the code will make much more sense if `query` is a single source of truth and thus always contains all the values that are actually used when sending query to api.
I don't think we should worry how the URL looks for the user - it looks ugly anyways since it's url-encoded 😅 

That being said this PR is blocking some other work, so I'm a fan of leaving this conversation for later and moving forward with this.

Reply:
I think the concern here is that we were gonna use the decoded URL to populate the search bar (once that's available) with the value the user input, but I see now that any interaction with the UI will also cause this issue. From the example above:

1. User types simple query like `type:expenses`
2. Press the date column name to change the sorting order.
3. Now the query looks like `type:expenses sortOrder:asc sortBy:date`
4. We'd display `type:expenses sortOrder:asc sortBy:date` in the Search bar, which is different than what the user typed
5. The user presses the column again and in theory we have same result as in the step 1 but the query looks differently `type:expenses sortOrder:desc sortBy:date`, which is also not what the user input was.

I think we might need another solution for this problem, so in all cases we only show `type:expenses` to the user since that's what they typed into the search bar. That's a problem for v2.3, but we should keep it in mind.